src/Symfony/Component/VarDumper/Caster/DateCaster.php

```diff
@@ -80,6 +82,41 @@ public static function castTimeZone(\DateTimeZone $timeZone, array $a, Stub $stu
         return $filter & Caster::EXCLUDE_VERBOSE ? $z : $z + $a;
     }
 
+    public static function castPeriod(\DatePeriod $p, array $a, Stub $stub, $isNested, $filter)
+    {
+        if (defined('HHVM_VERSION_ID') || \PHP_VERSION_ID < 50605) {
+            return $a;
+        }
+
+        $dates = array();
+        if (\PHP_VERSION_ID >= 70107) { // see https://bugs.php.net/bug.php?id=74639
+            foreach (clone $p as $i => $d) {
+                if (self::PERIOD_LIMIT === $i) {
+                    // avoids iterator_count() function which can be expensive
```

Comment:
not sure about the comment: nobody was wondering about iterator_count until read :)